### ISSUE TITLE
Weird crash issues

### ISSUE THREAD

**Issue by CliffracerX** [iChun/Tabula#31]
If you accidentally delete everything in one of the rotation boxes and then put in a single "-" symbol, the game crashes.

```java.lang.NumberFormatException: For input string: "-"
  	at sun.misc.FloatingDecimal.readJavaFormatString(Unknown Source)
  	at java.lang.Double.parseDouble(Unknown Source)
  	at us.ichun.mods.tabula.client.gui.window.WindowControls.elementTriggered(WindowControls.java:306)
  	at us.ichun.mods.tabula.client.gui.window.element.ElementNumberInput.keyInput(ElementNumberInput.java:339)
  	at us.ichun.mods.tabula.client.gui.GuiWorkspace.func_73869_a(GuiWorkspace.java:1497)
  	at net.minecraft.client.gui.GuiScreen.func_146282_l(GuiScreen.java:319)
  	at net.minecraft.client.gui.GuiScreen.func_146269_k(GuiScreen.java:276)
  	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1629)
  	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:962)
  	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:887)
  	at net.minecraft.client.main.Main.main(SourceFile:148)
  	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
  	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
  	at java.lang.reflect.Method.invoke(Unknown Source)
  	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
  	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)```

**Comment by laz2727** [comment]
Ugh. That's the exact same issue.

**Comment by CliffracerX** [comment]
Not quite, #17 was for empty strings, this is for me accidentally pressing
"-" instead of "0" and hitting enter.  Hypertyping is apparently a bad idea
with Tabula, lol

**Comment by laz2727** [comment]
Still, the issue is basically same - stupid things in random places cause
crashes. I bet Tabula crashes because of unexpected stuff in the model too.

**Comment by CliffracerX** [comment]
Yeah, although I'm not sure if this issue is on iChun's part or simply Java
flipping out when it can't parse something.  :/

**Comment by laz2727** [comment]
Crashes are ALWAYS creator's issue.

**Comment by iChun** [comment]
Unfortunately I'm on holiday and I've already started porting Tabula to 1.8, so Tabula for 1.7.10 is EOL, with 4.1.1 being the last version which I built before I left on holiday. Just try not to put in any -'s in number input boxes for the time being.

@laz2727 you don't have to be so mean, sure, the crashes are my fault, yes, but at least we have someone reporting the crashes and I am actually trying to fix them. You'd be surprised how many people do "stupid", unintended things.

**Comment by CliffracerX** [comment]
The whole mod EOL thing is a real quibble of mine, LexManos being one of
the bigger offenders of said quibble.

90% of the playerbase don't use 1.8 yet.  A good collection of people don't
even use 1.7 yet.  You should try to always give support to the version
before the latest, and then when there's another major update finally drop
support.  E.G, continue supporting 1.7 until 1.9 comes out, then when
2.0/1.11 or whatever it is comes out drop 1.8 support.

Just some thoughts, up to you if you want to use that idea.

**Comment by iChun** [comment]
The thing is, I've actually tried going down that path before, after all I've been modding for almost 4 years now. The problem is, it's a large waste of time to maintain so much of code, you'd be better off working on something more important, especially when big minecraft updates change the code so much. You'd be better off using that time working on more content that'll give players the push to move to the new version even. 

I've eventually formed just a general system of "if the mod is pretty new, with a new minecraft version just out, provide support for both versions, even consider downporting if the mod was made on the latest version. Support those for a while and when more so people move to the latest version of Minecraft, drop support for the old version. If the mod has been out for a while and a new MC version has been released, once the port begins, the mod is pretty much EOL for that MC version, if an issue is reported on the old MC version, see if it's still applicable in the new version and if it is, fix it there." Again like I said earlier, time is a thing here I don't really want to be wasting here... so... yeah
